projectmainpage.html

```diff
@@ -0,0 +1,265 @@
+                    <li><input type="checkbox" id="blush" name="blush"> Blush</li>
+                    <li><input type="checkbox" id="powder" name="powder"> Powder</li>
+                </ul>
+                <br>
+                <h3>Skin Tone Filters:</h3>
+                <ul>
+                    **This filter can ONLY be applied when foundation, concealer, and bronzer products are filtered**
+                </ul>
+                    <ul>
+                        <li><input type="radio" id="skin-tone-fair" name="skin-tone" value="fair"> Light</li>
+                        <li><input type="radio" id="skin-tone-medium" name="skin-tone" value="medium"> Medium</li>
+                        <li><input type="radio" id="skin-tone-dark" name="skin-tone" value="dark"> Dark</li>
+                        <li><input type="radio" id="skin-tone-none" name="skin-tone" value="none" checked> No Filter Applied</li>
+                    </ul>
+                <br>
+                <h3>Price Filters:</h3>
+                <ul>
+                    <li><input type="checkbox" id="under-10" name="under-10"> Under $10</li>
+                    <li><input type="checkbox" id="10-20" name="10-20"> $10 - $20</li>
+                    <li><input type="checkbox" id="20-50" name="20-50"> $20 - $50</li>
+                    <li><input type="checkbox" id="50-plus" name="50-plus"> $50 and above</li>
+                </ul>
+            </aside>
+
+            <!-- recommendations section with headers-->
+            <section id="home-recommendations">
+                <h2>Welcome to Makeup Manager!</h2>
+                <br>
+                <br>
+                <h2>Makeup reccomendations for you: </h2>
+                <br>
+                <br>
+                <div id="recommendation-grid"></div>
+            </section>
+        </main>
+
+        <br>
+        <!-- Footer -->
+        <footer id="footer">
+            <p>Copyright by MakeupManager</p>
+        </footer>
+
+        <script>
+            const recommendationsGrid = document.getElementById('recommendation-grid');
+            const filterContainer = document.getElementById('filter-container');
+            const searchBar = document.getElementById('search-bar');
+            const loadingPopup = document.getElementById('loading-popup');
+
+            let allProducts = [];
+            let filteredProducts = [];
+            let savedProducts = []; 
+
+            function showLoading() {
+                loadingPopup.style.display = 'flex';
+                loadingPopup.innerText = 'Loading recommendations, please wait...';
+            }
+
+            function hideLoading() {
+                loadingPopup.style.display = 'none';
+            }
+
+            showLoading();
+            fetch('http://makeup-api.herokuapp.com/api/v1/products.json')
+                .then(response => response.json())
+                .then(data => {
+                    allProducts = data.map(product => {
```

Comment:
We actually have the advantage of dividing the work into a few small asynchronous requests in the form

`GET http://makeup-api.herokuapp.com/api/v1/products.json?product_type=foundation`
`GET http://makeup-api.herokuapp.com/api/v1/products.json?product_type=concealer`
`GET http://makeup-api.herokuapp.com/api/v1/products.json?product_type=bronzer`